tests/modules/test_k8s_prerequisites_check.py

```diff
@@ -0,0 +1,85 @@
+import unittest
+from unittest.mock import Mock
+
+from rookify.modules.exception import ModuleException
+from typing import Any, Dict, List
+from ..mock_k8s_prerequisite_check import MockK8sPrerequisitesCheckHandler
+
+
+# Note: currently this test works with pytest but not with unittest, which is not able to import needed classes
+class TestK8sPrerequisitesCheckHandler(unittest.TestCase):
+    def setUp(self) -> None:
+        # Mock configuration
+        self.config = {"rook": {"cluster": {"namespace": "test-namespace"}}}
+        # No response option
+        self.empty_response = False
+
+    def _request_callback(
+        self, method: str, *args: List[Any], **kwargs: Dict[Any, Any]
+    ) -> Any:
+        if method == "apps_v1_api.list_deployment_for_all_namespaces":
+            if self.empty_response is True:
```

Comment:
You may want to use a more descriptive name for this flag to differentiate test calls for the deployment list.

Reply:
thanks good idea, changed flag to empty_deployment_list